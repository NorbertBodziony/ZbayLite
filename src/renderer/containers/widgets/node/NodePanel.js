import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelComponent from '../../../components/widgets/node/NodePanel'
import nodeHandlers from '../../../store/handlers/node'
import { useInterval } from '../../hooks'

export const mapDispatchToProps = dispatch => bindActionCreators({
  getStatus: nodeHandlers.epics.getStatus
}, dispatch)

export const NodePanel = ({ className, getStatus }) => {
  useInterval(getStatus, 15000)
  return (
    <NodePanelComponent className={className} />
  )
}

export default connect(null, mapDispatchToProps)(NodePanel)
