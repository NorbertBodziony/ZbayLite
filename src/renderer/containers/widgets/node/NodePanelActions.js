import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import NodePanelActions from '../../../components/widgets/node/NodePanelActions'

import nodeHandlers from '../../../store/handlers/node'

export const mapDispatchToProps = dispatch => bindActionCreators({
  onRestart: nodeHandlers.epics.restart,
  onPower: nodeHandlers.epics.togglePower
}, dispatch)

export default connect(null, mapDispatchToProps)(NodePanelActions)
