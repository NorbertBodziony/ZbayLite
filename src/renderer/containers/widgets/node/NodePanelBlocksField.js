import React from 'react'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  latestBlock: nodeSelectors.latestBlock(state),
  currentBlock: nodeSelectors.currentBlock(state)
})

export const NodePanelBlocksField = ({ latestBlock, currentBlock }) => (
  <NodePanelField name='Blocks'>
    <Typography inline variant='overline'>
      {currentBlock} / {latestBlock || '?'}
    </Typography>
  </NodePanelField>
)

export default connect(mapStateToProps)(NodePanelBlocksField)
