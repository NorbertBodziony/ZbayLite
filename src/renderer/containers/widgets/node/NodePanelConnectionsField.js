import React from 'react'
import BigNumber from 'bignumber.js'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  connections: nodeSelectors.connections(state)
})

export const NodePanelConnectionsField = ({ connections }) => (
  <NodePanelField name='Connections'>
    <Typography inline variant='overline'>
      {connections.toString()}
    </Typography>
  </NodePanelField>
)

NodePanelConnectionsField.propTypes = {
  connections: PropTypes.instanceOf(BigNumber).isRequired
}

export default connect(mapStateToProps)(NodePanelConnectionsField)
