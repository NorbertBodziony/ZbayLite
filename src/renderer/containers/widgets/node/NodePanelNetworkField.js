import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  network: nodeSelectors.network(state)
})

export const NodePanelNetworkField = ({ network }) => (
  <NodePanelField name='Network'>
    <Typography inline variant='overline'>
      {network || '?'}
    </Typography>
  </NodePanelField>
)

NodePanelNetworkField.propTypes = {
  network: PropTypes.oneOf(['mainnet', 'testnet'])
}

export default connect(mapStateToProps)(NodePanelNetworkField)
