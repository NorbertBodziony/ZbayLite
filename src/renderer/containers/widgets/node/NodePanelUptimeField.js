import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import { connect } from 'react-redux'

import Typography from '@material-ui/core/Typography'

import NodePanelField from '../../../components/widgets/node/NodePanelField'

import nodeSelectors from '../../../store/selectors/node'

export const mapStateToProps = state => ({
  uptime: nodeSelectors.uptime(state)
})

export const NodePanelUptimeField = ({ uptime }) => (
  <NodePanelField name='Uptime'>
    <Typography display='inline' variant='overline'>
      {R.isEmpty(uptime) ? '-' : null}
      {uptime.days ? `${uptime.days}d  ` : null}
      {uptime.hours ? `${uptime.hours}h  ` : null}
      {uptime.minutes ? `${uptime.minutes}m` : null}
    </Typography>
  </NodePanelField>
)

NodePanelUptimeField.propTypes = {
  uptime: PropTypes.exact({
    days: PropTypes.number,
    hours: PropTypes.number,
    minutes: PropTypes.number,
    seconds: PropTypes.number
  }).isRequired
}

export default connect(mapStateToProps)(NodePanelUptimeField)
