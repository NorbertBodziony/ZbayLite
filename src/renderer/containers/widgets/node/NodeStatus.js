import { connect } from 'react-redux'

import NodeStatus from '../../../components/widgets/node/NodeStatus'

import nodeSelectors from '../../../store/selectors/node'
import identitySelectors from '../../../store/selectors/identity'

export const mapStateToProps = state => ({
  status: nodeSelectors.status(state),
  percentSynced: nodeSelectors.percentSynced(state),
  freeUtxos: identitySelectors.freeUtxos(state)
})

export default connect(mapStateToProps)(NodeStatus)
