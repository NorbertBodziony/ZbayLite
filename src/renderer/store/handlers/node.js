import Immutable from 'immutable'
import BigNumber from 'bignumber.js'
import { createAction, handleActions } from 'redux-actions'

import { typeRejected, LoaderState } from './utils'
import { getClient } from '../../zcash'

const DEFAULT_ADDRESS_TYPE = 'sapling'

export const NodeState = Immutable.Record({
  latestBlock: new BigNumber(0),
  currentBlock: new BigNumber(0),
  connections: new BigNumber(0),
  isTestnet: null,
  status: 'connecting',
  errors: '',
  bootstrapLoader: LoaderState(),
  startedAt: null
}, 'NodeState')

export const initialState = NodeState()

export const actionTypes = {
  GET_STATUS: 'GET_NODE_STATUS',
  CREATE_ADDRESS: 'CREATE_ZCASH_ADDRESS',
  GET_BALANCE: 'GET_BALANCE',
  SET_STATUS: 'SET_NODE_STATUS',
  SET_BOOTSTRAPPING: 'SET_NODE_BOOTSTRAPPING',
  SET_BOOTSTRAPPING_MESSAGE: 'SET_NODE_BOOTSTRAPPING_MESSAGE'
}

const setStatus = createAction(actionTypes.SET_STATUS)

const createAddress = createAction(
  actionTypes.CREATE_ADDRESS,
  async (type = DEFAULT_ADDRESS_TYPE) => {
    return getClient().addresses.create(type)
  }
)

const setBootstrapping = createAction(actionTypes.SET_BOOTSTRAPPING)
const setBootstrappingMessage = createAction(actionTypes.SET_BOOTSTRAPPING_MESSAGE)

const actions = {
  createAddress,
  setBootstrapping,
  setBootstrappingMessage
}

const getStatus = () => async (dispatch) => {
  try {
    const info = await getClient().status.info()
    dispatch(setStatus(info))
  } catch (err) {
    dispatch(setStatus({ 'status': 'down', errors: err }))
  }
}

const restart = () => (dispatch) => {
  console.log('Restarting node')
}

const togglePower = () => (dispatch) => {
  console.log('toggling power of the node')
}

const epics = {
  getStatus,
  restart,
  togglePower
}

export const reducer = handleActions({
  [actionTypes.SET_STATUS]: (state, { payload: status }) => state.merge(status),
  [typeRejected(actionTypes.CREATE_ADDRESS)]: (state, { payload: errors }) => state.set('errors', errors),
  [setBootstrapping]: (state, { payload: bootstrapping }) => state.setIn(['bootstrapLoader', 'loading'], bootstrapping),
  [setBootstrappingMessage]: (state, { payload: message }) => state.setIn(['bootstrapLoader', 'message'], message)
}, initialState)

export default {
  actions,
  reducer,
  epics
}
