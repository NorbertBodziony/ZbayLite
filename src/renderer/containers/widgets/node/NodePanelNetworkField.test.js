/* eslint import/first: 0 */
jest.mock('../../../vault')
import React from 'react'
import Immutable from 'immutable'
import { shallow } from 'enzyme'

import { mapStateToProps, NodePanelNetworkField } from './NodePanelNetworkField'

import create from '../../../store/create'
import { NodeState } from '../../../store/handlers/node'

describe('NodePanelNetworkField', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      initialState: Immutable.Map({
        node: NodeState({
          isTestnet: true
        })
      })
    })
  })

  it('will receive right props', async () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })

  it('renders the field', async () => {
    const result = shallow(
      <NodePanelNetworkField
        {...mapStateToProps(store.getState())}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
