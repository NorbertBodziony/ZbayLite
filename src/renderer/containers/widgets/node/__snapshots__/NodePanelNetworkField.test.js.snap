// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodePanelNetworkField renders the field 1`] = `
<Memo(WithStyles(NodePanelField))
  name="Network"
  value="testnet"
/>
`;

exports[`NodePanelNetworkField renders the field when no network 1`] = `
<Memo(WithStyles(NodePanelField))
  name="Network"
  value={null}
/>
`;

exports[`NodePanelNetworkField will receive right props 1`] = `
Object {
  "network": "testnet",
}
`;
